app/models/user.rb

```diff
@@ -9,16 +9,52 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
-  has_many :senders, foreign_key: 'sender_id', class_name: 'Friendship', dependent: :destroy
-  has_many :receivers, foreign_key: 'receiver_id', class_name: 'Friendship', dependent: :destroy
 
-  def sender_friends_id
-    senders.where('status = ? AND (sender_id = ? OR receiver_id = ?)', 1, id, id).pluck(:sender_id,
-                                                                                        :receiver_id).flatten
+  has_many :invitations_received, foreign_key: :invitation_receiver_id, class_name: 'Friendship', dependent: :destroy
+  has_many :users_who_invited_me, through: :invitations_received, source: :invitation_sender, dependent: :destroy
+
+  has_many :invitations_sent, foreign_key: :invitation_sender_id, class_name: 'Friendship', dependent: :destroy
+  has_many :users_invited_by_me, through: :invitations_sent, source: :invitation_receiver, dependent: :destroy
+
+  def friendship_request(user)
+    return false if users_invited_by_me.include?(user) || users_who_invited_me.include?(user)
+
+    users_invited_by_me << user
+  end
+
+  def accept_request(user)
+    friend_request = invitations_received.where(invitation_sender_id: user.id).first
+    friend_request.update(status: true)
+  end
+
+  def reject_request(user)
+    friend_request = invitations_received.where(invitation_sender_id: user.id).first
+    friend_request.destroy
+  end
+
+  def friends
+    friend_sender = User.where(id: invitations_sent.where(status: true).pluck(:invitation_receiver_id))
+    friend_receiver = User.where(id: invitations_received.where(status: true).pluck(:invitation_sender_id))
+    friend_sender + friend_receiver
+  end
```

Comment:
- [ ] Once you update the friendship model to record a reverse of every friendship, you can take advantage of it to get friends of a user by using one column (user_id)  with a status of true. Currently, you are checking both columns(user_id and friend_id), which results in a greater number of queries and performs less efficiently.